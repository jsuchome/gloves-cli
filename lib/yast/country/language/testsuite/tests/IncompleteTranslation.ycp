/**
 * tests for Language::IncompleteTranslation
 * Author:	jsuchome@suse.cz
 * $Id: IncompleteTranslation.ycp 31976 2006-07-25 08:02:24Z jsuchome $
 */
{

    // testedfiles: Language.ycp

    include "testsuite.ycp";

    map READ = $[
	"target" : $[
	    "bash_output"	: $[],
	    "size"		: 1,
	    "yast2"		: $[],
	],
    ];

    TESTSUITE_INIT([READ,$[],READ], nil);

    map R	= $[
	"target" : $[
	    "string"	: "",
	    "stat"	: $[],
	]
    ];

    import "Language";

    // not present	-> complete
    TEST(``(Language::IncompleteTranslation ("en_US")), [R,$[],$[]], nil);

    // incomplete
    R["target","string"]	= "50";
    R["target","stat"]		= $[ 1:2];
    TEST(``(Language::IncompleteTranslation ("en_GB")), [R,$[],$[]], nil);

    // no SCR call, cached from previous
    TEST(``(Language::IncompleteTranslation ("en_GB")), [R,$[],$[]], nil);

    // complete
    R["target","string"]	= "99";
    TEST(``(Language::IncompleteTranslation ("de_DE")), [R,$[],$[]], nil);


}
