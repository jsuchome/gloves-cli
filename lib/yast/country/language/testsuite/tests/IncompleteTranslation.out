Read	.target.size "/etc/sysconfig/language" 1
Read	.target.string "/usr/lib/YaST2/trans/en.status" ""
Return	false
Read	.target.string "/usr/lib/YaST2/trans/en_GB.status" "50"
Return	true
Return	true
Read	.target.string "/usr/lib/YaST2/trans/de_DE.status" "99"
Return	false
